lib/rspec-puppet/facter_impl.rb

```diff
@@ -0,0 +1,39 @@
+module RSpec::Puppet
+  class FacterTestImpl
+    def initialize
+      @facts = {}
+    end
+
+    def value(fact_name)
+      @facts[fact_name.to_s]
+    end
+
+    def debugging(arg); end
+
+    def reset; end
+
+    def search(*paths); end
+
+    def setup_logging; end
+
+    def clear
+      @facts.clear
+    end
+
+    def setcode(string = nil, &block)
+      if block_given?
+        value = block.call.to_s
```

Comment:
```suggestion
        value = block.call
```

this might mess with facts that have hashes as values, though I don't know why this case didn't surface while testing